qt/widgets/instrumentview/src/InstrumentActor.cpp

```diff
@@ -86,45 +103,53 @@ InstrumentActor::InstrumentActor(const QString &wsName, bool autoscaling,
   // set up data ranges and colours
   setUpWorkspace(sharedWorkspace, scaleMin, scaleMax);
 
-  Instrument_const_sptr instrument = getInstrument();
-
   // If the instrument is empty, maybe only having the sample and source
-  const int nelements = instrument->nelements();
-  if ((nelements == 0) || (nelements == 1 && (instrument->getSource() ||
-                                              instrument->getSample())) ||
-      (nelements == 2 && instrument->getSource() && instrument->getSample())) {
+  auto nelements = componentInfo.size();
+  if ((nelements == 0) || (nelements == 1 && (componentInfo.hasSource() ||
+                                              componentInfo.hasSample())) ||
+      (nelements == 2 && componentInfo.hasSource() &&
+       componentInfo.hasSample())) {
```

Comment:
Couldn't it also have no detectors, even if there were a large number of non-detectors? The one thing not checked here is the actual number of detectors.